### PR TITLE
Update patches to handle vineflower decompiler

### PR DESCRIPTION
**Pull request by Machine-Maker**

### CODE REVIEW

File: patches/server/0202-Add-EntityKnockbackByEntityEvent-and-EntityPushedByE.patch [hunk 1]
```diff
@@ -85,29 +85,29 @@ index 21bef887193c777ec1e311b3bf5c0213180ea4ca..83a56061ce23d3d6f7d8ad3201ce3251
...
-                 target.hurt(world.damageSources().sonicBoom(entity), 10.0F);
-                 double d = 0.5D * (1.0D - target.getAttributeValue(Attributes.KNOCKBACK_RESISTANCE));
-                 double e = 2.5D * (1.0D - target.getAttributeValue(Attributes.KNOCKBACK_RESISTANCE));
--                target.push(vec33.x() * e, vec33.y() * d, vec33.z() * e);
+@@ -81,7 +81,7 @@ public class SonicBoom extends Behavior<Warden> {
+                     target.hurt(world.damageSources().sonicBoom(entity), 10.0F);
+                     double d = 0.5 * (1.0 - target.getAttributeValue(Attributes.KNOCKBACK_RESISTANCE));
+                     double e = 2.5 * (1.0 - target.getAttributeValue(Attributes.KNOCKBACK_RESISTANCE));
+-                    target.push(vec33.x() * e, vec33.y() * d, vec33.z() * e);
 +                target.push(vec33.x() * e, vec33.y() * d, vec33.z() * e, entity); // Paper - Add EntityKnockbackByEntityEvent and EntityPushedByEntityAttackEvent
```
Comment (jpenilla) [inline patches/server/0202-Add-EntityKnockbackByEntityEvent-and-EntityPushedByE.patch]: this looks like it needs indent

File: patches/server/0245-Improve-death-events.patch [hunk 2]
```diff
@@ -319,30 +319,27 @@ index 6376908df89af1eff3a948ca1faef5d4925f0c3b..4ac32078cb074a78a2bb83307013b849
...
+index e9137ea77a44eb8ded24a70f3a2b388d77379fa7..ecf8a1ddf4a98bc3587708975c4045e02ed09b38 100644
 --- a/src/main/java/net/minecraft/world/entity/animal/horse/AbstractChestedHorse.java
 +++ b/src/main/java/net/minecraft/world/entity/animal/horse/AbstractChestedHorse.java
-@@ -70,11 +70,19 @@ public abstract class AbstractChestedHorse extends AbstractHorse {
+@@ -70,9 +70,17 @@ public abstract class AbstractChestedHorse extends AbstractHorse {
                  this.spawnAtLocation(Blocks.CHEST);
              }
  
--            this.setChest(false);
 +            //this.setCarryingChest(false); // Paper - moved to post death logic
```
Comment (jpenilla) [inline patches/server/0245-Improve-death-events.patch]: maybe update this comment while we are here

File: patches/server/0407-Fix-AdvancementDataPlayer-leak-due-from-quitting-ear.patch [hunk 3]
```diff
@@ -21,10 +21,8 @@ index 5450734060428426981e2c1f1025b1aa5aa4d271..805459ba815520165c86a537fbb1f3e0
  
      @Override
      public final void addPlayerListener(PlayerAdvancements manager, CriterionTrigger.Listener<T> conditions) {
--        this.players.computeIfAbsent(manager, (managerx) -> {
-+        manager.criterionData.computeIfAbsent(this, (managerx) -> { // Paper - fix AdvancementDataPlayer leak
-             return Sets.newHashSet();
-         }).add(conditions);
+-        this.players.computeIfAbsent(manager, managerx -> Sets.newHashSet()).add(conditions);
++        manager.criterionData.computeIfAbsent(this, managerx -> Sets.newHashSet()).add(conditions);
```
Comment (kennytv) [inline patches/server/0407-Fix-AdvancementDataPlayer-leak-due-from-quitting-ear.patch]: Missing comment

File: patches/server/0441-Reduce-blockpos-allocation-from-pathfinding.patch [hunk 4]
```diff
@@ -17,12 +17,12 @@ index dbb6a7d950b1a6f58a845b4d399ead6eb6d4fe14..70c7c735fdd0c1b66243915622e47eab
...
-                 } else if (blockState.is(Blocks.COCOA)) {
-                     return BlockPathTypes.COCOA;
-                 } else if (!blockState.is(Blocks.WITHER_ROSE) && !blockState.is(Blocks.POINTED_DRIPSTONE)) {
--                    FluidState fluidState = world.getFluidState(pos);
+@@ -516,7 +516,7 @@ public class WalkNodeEvaluator extends NodeEvaluator {
+         } else if (blockState.is(Blocks.COCOA)) {
+             return BlockPathTypes.COCOA;
+         } else if (!blockState.is(Blocks.WITHER_ROSE) && !blockState.is(Blocks.POINTED_DRIPSTONE)) {
+-            FluidState fluidState = world.getFluidState(pos);
 +                    FluidState fluidState = blockState.getFluidState(); // Paper - Perf: Reduce blockpos allocation from pathfinding
```
Comment (jpenilla) [inline patches/server/0441-Reduce-blockpos-allocation-from-pathfinding.patch]: indent looks wrong here

File: patches/server/1013-Use-Velocity-compression-and-cipher-natives.patch [hunk 5]
```diff
@@ -229,10 +229,9 @@ index 859af8c845bae9781a62fa4acae56c6e2d449e10..f67f59f287d9a5cdd685b6b56ed1daf3
 +                compatibleIn.release();
 +            }
 +            // Paper end - Use Velocity cipher
-         }
- 
-     }
- 
++        }
```
Comment (jpenilla) [inline patches/server/1013-Use-Velocity-compression-and-cipher-natives.patch:233]: looks like start/end comments might need to be moved here?

File: patches/server/1034-Improve-tag-parser-handling.patch [hunk 6]
```diff
@@ -64,55 +64,55 @@ index 5bec54239a2b185284c10d58854e5a13e33daae5..ef970428fe9fe36c242ce6588bb6843e
...
-+                    list.add((T)((NumericTag)tag).getAsNumber()); // Paper - decompile fix
-                 } else if (typeReader == LongTag.TYPE) {
--                    list.add((T)((NumericTag)tag).getAsLong());
-+                    list.add((T)((NumericTag)tag).getAsNumber()); // Paper - decompile fix
-                 } else {
--                    list.add((T)((NumericTag)tag).getAsInt());
-+                    list.add((T)((NumericTag)tag).getAsNumber()); // Paper - decompile fix
-                 }
+             if (typeReader == ByteTag.TYPE) {
+-                list.add((T)((NumericTag)tag).getAsByte());
```
Comment (jpenilla) [inline patches/server/1034-Improve-tag-parser-handling.patch]: should these be moved to mcdev fixes?

File: patches/server/1008-Optimise-general-POI-access.patch [hunk 7]
```diff
@@ -894,58 +899,67 @@ index 6f2c7baea0d1ac7813c7b85e1f5558573745762c..12a7aaeaa8b4b788b620b1985591c3b9
...
 +            this, typePredicate, biPredicate, pos, radius, radius * radius, Occupancy.HAS_SPACE, false
 +        );
 +        return Optional.ofNullable(closest).map(poi -> {
 +            // Paper end - re-route to faster logic
-             poi.acquireTicket();
-             return poi.getPos();
-         });
++            poi.acquireTicket();
++            return poi.getPos();
++        });
```
Comment (Maintenancer) [inline patches/server/1008-Optimise-general-POI-access.patch]: move .map down a line and put the Paper end commend above that (and revert indent change on these 3 lines)

File: patches/server/1015-Collision-optimisations.patch [hunk 8]
```diff
@@ -2466,18 +2466,18 @@ index ffa4f34d964fbcc53e2dfe11677832db21a6eb93..7618364e5373fe17cfe45a5a4ee9ab25
...
-+                    if (!io.papermc.paper.util.CollisionUtil.voxelShapeIntersect(this.box, (double)i, (double)j, (double)k, (double)i + 1.0D, (double)j + 1.0D, (double)k + 1.0D)) { // Paper - keep vanilla behavior for voxelshape intersection - See comment in CollisionUtil
-                         continue;
-                     }
- 
+@@ -99,7 +99,7 @@ public class BlockCollisions<T> extends AbstractIterator<T> {
+                         // Paper end
+                         VoxelShape voxelShape = blockState.getCollisionShape(this.collisionGetter, this.pos, this.context);
+                         if (voxelShape == Shapes.block()) {
+-                            if (this.box.intersects((double)i, (double)j, (double)k, (double)i + 1.0, (double)j + 1.0, (double)k + 1.0)) {
++                            if (!io.papermc.paper.util.CollisionUtil.voxelShapeIntersect(this.box, (double)i, (double)j, (double)k, (double)i + 1.0D, (double)j + 1.0D, (double)k + 1.0D)) { // Paper - keep vanilla behavior for voxelshape intersection - See comment in CollisionUtil
```
Comment (kennytv) [inline patches/server/1015-Collision-optimisations.patch]: Bad negation

**Review by kennytv — CHANGES_REQUESTED** [review]

**Review by kennytv — APPROVED** [review]
Should start a server and see if things around collision work as expected, but otherwise should be good

**Review by lynxplay — APPROVED** [review]

**Review by jpenilla — APPROVED** [review]

**Pull request merged**